### DISCUSSION TITLE
New audit: detecting potentially blasted caches (cache poisoning)

### DISCUSSION THREAD

**Discussion by ubiratansoares** [zizmorcore/zizmor#186] [Ideas]
I was checking the [roadmap](https://github.com/woodruffw/zizmor/issues/1) yesterday and got to know about [GHA cache poisoning attacks](https://adnanthekhan.com/2024/05/06/the-monsters-in-your-build-cache-github-actions-cache-poisoning/). Always learning, as usual 😄

I was wondering how could we add an initial support to this audit on `zizmor`. My first line of thought was simple : 

- Start only with Github-provided actions with built-in caching(eg, the `actions/cache/restore`, `actions/setup-java`, etc)
- Detected dangerouns conditions attached to cache restoration (usually Workflow triggers, but eventually something else from the CI context) 
- Assign a low confidence to anything we find
- Warn that caches are enabled by default and should be disabled when under those conditions

[I got in touch with Adnan](https://github.com/AdnaneKhan/ActionsCacheBlasting/issues/3) (from the blog post above), he drove a more generic recommendation to stay safe, but in the scope of `zizmor`, perphaps we can start simple and improve over time, eg by having a curated list of popular Actions built on top of `actions/toolkit`  cache functionality.

@woodruffw Wdyt?

**Comment by woodruffw** [comment]
Thanks @ubiratansoares! Yeah, I think it would be great to begin experimenting here.

As you pointed out, a tricky thing with analyzing for cache poisoning is sensitivity: `zizmor` doesn't currently have enough context/detail in the analyses it does to say confidently that that a given workflow *is* vulnerable to cache poisoning, just *potentially* vulerable. 

Given that, I *think* I'm OK with us having a pedantic-only audit for dangerous triggers + cache use for the time being, but I'd love it if we could more generally improve our analysis of dangerous-trigger workflows to be more confident that they're actually exploitable.

TL;DR: I'm good with adding this audit, but we should probably have it be pedantic-only for now. Alternatively, we could hold off on writing this audit and continue working on related audits that provide more context for dangerous triggers (like the `GITHUB_ENV` audit).

**Reply by ubiratansoares** [reply]
`pedantic` looks like the right fit, I totally agree with you! 

I also agree that other audits can come first, happy to take another of them 🔥

**Comment by woodruffw** [comment]
I've broken this out into an issue: #261, so closing here! Thanks @ubiratansoares!
